basic_pipeline/09.0_Pipeline.md

```diff
@@ -0,0 +1,133 @@
+  link(:input2) |> to(:ordering_buffer2) |> to(:depayloader2),
+  link(:depayloader1) |> via_in(:first_input) |> to(:mixer),
+  link(:depayloader2) |> via_in(:second_input) |> to(:mixer),
+  link(:mixer) |> to(:output)
+ ]
+ ...
+end
+```
+We hope the syntax is visually descriptive enough to show what is the desired result of that definition. But to make sure everything is clear to you let me tell you something about it.
+Each pipeline's "branch" starts with the `link/1` which takes as an argument an atom corresponding to a given element. All the further elements in the branch can be accessed with the use of the `to/1` function, expecting an atom that identifies that element to be passed as an argument. Note, that the mentioned atoms must be the same as the ones you have used as keys in the `children` map!
+`|>` operator allows "linking" of the elements accessed in the way described above, via their pads. By default, the elements' link will be using a `:input` pad as the input pad and a `:output` pad as the output pad.
+`via_in/1` allows specifying an input pad with a given name. Since in a mixer there are two input pads (`:first_input` and `:second_input`, defined in `Basic.Elements.Mixer` module with `def_input_pad` and `def_output_pad` macros), we need to distinguish between them while linking the elements. 
+Of course, there is also a `via_out/1` function, which is used to point the output pad with the given identifier, but there was no need to use it. 
+In the case of other elements we do not need to explicitly point the desired pads since we are taking advantage of the default pads name - `:input` for the input pads and `:output` for the output ones (see what names we have given to our pads in the elements other than the mixer!). However, we could rewrite the following `links` definitions and explicitly specify the pad names:
+```Elixir
+# FILE: lib/Pipeline.ex
+
+def handle_init(_opts) do
+ ...
+ links = [
+  link(:input1) |> via_out(:output) |> via_in(:input) |> to(:ordering_buffer1) |> via_out(:output) |> via_in(:input) |> to(:depayloader1),
+  link(:input2) |> via_out(:output) |> via_in(:input) |> to(:ordering_buffer2) |> via_out(:output) |> via_in(:input) |>to(:depayloader2),
+  link(:depayloader1) |> via_out(:output) |> via_in(:first_input) |> to(:mixer),
+  link(:depayloader2) |> via_out(:output) |> via_in(:second_input) |> to(:mixer),
+  link(:mixer) |> |> via_out(:output) |> via_in(:input) |> to(:output)
+ ]
+ ...
+end
+```
+
+That's almost it! All we need to do is to return a proper tuple from the `handle_init/1` callback, with the `:spec` action defined:
+```Elixir
+# FILE: lib/Pipeline.ex
+
+def handle_init(_opts) do
+ ...
+ spec = %ParentSpec{children: children, links: links}
+ {{:ok, spec: spec}, %{}}
+end
+```
+
+Our pipeline is ready! Let's try to launch it.
+You can do it by running an `iex` command with the `-S mix` option specified, which will make modules from our mix project available in the terminal.
```

Comment:
having thought about it for a while, I think it would be best to create `start.exs` file, which would start the pipeline:
```
{:ok, pid} = Basic.Pipeline.start()
Basic.Pipeline.play(pid)
Process.sleep(1000)
```
Note, that the sleep at the end is needed for the pipeline to finish.

The script would make it fast to run the pipeline without having to type the commands by hand each time.

Reply:
```
{:ok,  pid} = Basic.Pipeline.start()
Basic.Pipeline.play(pid)
Membrane.Pipeline.stop_and_terminate(pid, blocking?: true)
```